examples.md

```diff
@@ -381,17 +381,15 @@ Replace `~/.local/share/renv` with the correct `path` if not using Ubuntu.
 - uses: actions/cache@v1
   with:
     path: vendor/bundle
-    key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+    key: ${{ runner.os }}-gems-${{ hashFiles('.ruby-version') }}-${{ hashFiles('**/Gemfile.lock') }}
     restore-keys: |
-      ${{ runner.os }}-gems-
```

Comment:
Wouldn’t it be better to still restore the cache even if the Ruby version doesn’t match? As far as I understand, Bundler would then just rebuild gems as necessary and the cache would still help speed up installation.

Reply:
If there are no issues with pulling gems from a different ruby version, then we should have both `restore-keys`

Reply:
I've certainly seen bundler do nothing when native gems were installed for ruby 2.6.3, and then I upgraded to ruby 2.6.6. Perhaps ruby 2.6.3 and ruby 2.6.6 should be binary compatible and that's why bundler didn't re-install the gem, I'm not sure. That's what prompted the PR

Reply:
Peeking into the `vendor/bundle` folder, it seems like it’s partitioned based on Ruby’s minor version. For example, for Ruby 2.7.1, I have a `ruby/2.7.0` folder. I’m guessing this is because Ruby tries to maintain ABI compatibility within the same minor releases.

There could certainly be cases where gems need to be rebuilt and some might need to be rebuilt even if you don’t update Ruby (perhaps they’re linking some system library). Out of curiosity, could you name the gem that crashes under 2.6.6 when compiled against 2.6.3?